exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -1047,11 +1927,116 @@ func (e *Exchange) GetCurrencyTradeURL(_ context.Context, a asset.Item, cp curre
 	switch a {
 	case asset.Spot:
 		cp.Delimiter = currency.UnderscoreDelimiter
-		return poloniexAPIURL + tradeSpot + cp.Upper().String(), nil
+		return apiURL + tradeSpotPath + cp.Upper().String(), nil
 	case asset.Futures:
 		cp.Delimiter = ""
-		return poloniexAPIURL + tradeFutures + cp.Upper().String(), nil
+		return apiURL + tradeFuturesPath + cp.Upper().String(), nil
 	default:
-		return "", fmt.Errorf("%w %q", asset.ErrNotSupported, a)
+		return "", fmt.Errorf("%w: asset type: %v", asset.ErrNotSupported, a)
+	}
+}
+
+// WebsocketSubmitOrder submits an order to the exchange via a websocket connection
+func (e *Exchange) WebsocketSubmitOrder(ctx context.Context, s *order.Submit) (*order.SubmitResponse, error) {
+	var err error
+	s.Pair, err = e.FormatExchangeCurrency(s.Pair, s.AssetType)
+	if err != nil {
+		return nil, err
+	}
+	oTypeString, err := OrderTypeString(s.Type)
+	if err != nil {
+		return nil, err
+	}
+	var tif string
+	tif, err = TimeInForceString(s.TimeInForce)
+	if err != nil {
+		return nil, err
+	}
+	if s.AssetType != asset.Spot {
+		return nil, fmt.Errorf("%w: websocket order submit is not supported for asset type: %v", asset.ErrNotSupported, s.AssetType)
```

Comment:
🚧 Change request:
* Note, This isn't me just copy/pasting. I really don't think we should special case the error for just this, and the message is implicit.
```suggestion
		return nil, fmt.Errorf("%w: %q", asset.ErrNotSupported, s.AssetType)
```